src/utils/mesh_utils.ts

```diff
@@ -9,30 +9,71 @@ import {
     ValidationResults,
     WSClient,
 } from '@0x/mesh-rpc-client';
+import axios from 'axios';
 import * as _ from 'lodash';
 
-import { ZERO } from '../constants';
+import {
+    MESH_HTTP_DEFAULT_ORDER_BATCH_SIZE,
+    MESH_WEBSOCKET_DEFAULT_ORDER_BATCH_SIZE,
+    MESH_WEBSOCKET_ORDER_BATCH_LIMIT,
+    ZERO,
+} from '../constants';
 import { ValidationErrorCodes } from '../errors';
 import { logger } from '../logger';
 import { AddedRemovedUpdate, APIOrderWithMetaData } from '../types';
 
+import { utils } from './utils';
+
 export const meshUtils = {
     addOrdersToMeshAsync: async (
+        meshHttpEndpoint: string,
         meshClient: WSClient,
         orders: SignedOrder[],
-        batchSize: number = 100,
     ): Promise<ValidationResults> => {
-        // Mesh rpc client can't handle a large amount of orders. This results in a fragmented
-        // send which Mesh cannot accept.
+        // Mesh rpc client can't handle a large amount of orders over websocket. This results in a fragmented
+        // send which Mesh cannot accept. We check how many orders are being added, and use HTTP if there are
+        // too many.
+        const shouldSendViaHttp = orders.length > MESH_WEBSOCKET_ORDER_BATCH_LIMIT;
+        const batchSize = shouldSendViaHttp
+            ? MESH_HTTP_DEFAULT_ORDER_BATCH_SIZE
+            : MESH_WEBSOCKET_DEFAULT_ORDER_BATCH_SIZE;
         const validationResults: ValidationResults = { accepted: [], rejected: [] };
         const chunks = _.chunk(orders, batchSize);
         for (const chunk of chunks) {
-            const results = await meshClient.addOrdersAsync(chunk as any);
+            const results = shouldSendViaHttp
+                ? await meshUtils.addOrdersViaHttp(meshHttpEndpoint, chunk)
+                : await meshClient.addOrdersAsync(chunk);
             validationResults.accepted = [...validationResults.accepted, ...results.accepted];
             validationResults.rejected = [...validationResults.rejected, ...results.rejected];
         }
         return validationResults;
     },
+    // this logic mirrors the mesh RPC client and WebsocketProvider implementations
+    addOrdersViaHttp: async (
+        meshHttpEndpoint: string,
+        orders: SignedOrder[],
+        pinned: boolean = true,
```

Comment:
Should be set to `false`. Anyone can submit orders to 0x API and if we pin them, someone could spam our Mesh node and cause our DB to fill up and stop accepting new orders. Our anti-ddos/spam prevention only acts on orders that are not pinned.

Once we implement 0x API keys, we can pin orders from trusted makers. But until then, we shouldn't pin any of them.